extensions/amp-story/1.0/amp-story-share.js

```diff
@@ -277,13 +321,40 @@ export class ShareWidget {
     // TODO(alanorozco): Listen for proper tap event (i.e. fastclick)
     listen(linkShareButton, 'click', e => {
       e.preventDefault();
-      this.copyUrlToClipboard_();
+      this.copyUrlToClipboard_(this.root.querySelector('#page-share').checked);
     });
   }
 
-  /** @private */
-  copyUrlToClipboard_() {
-    const url = Services.documentInfoForDoc(this.getAmpDoc_()).canonicalUrl;
+  /**
+   * On desktop mode, with branching, users should be able to share a story
+   * starting from a specific page.
+   * @private
+  */
+  maybeAddPageShareButton_() {
+    const uiState = this.storeService_.get(StateProperty.UI_STATE);
+    const isDesktopUi =
+        uiState === UIType.DESKTOP_FULLBLEED ||
+        uiState === UIType.DESKTOP_PANELS;
+
+    if (isExperimentOn(this.win, 'amp-story-branching') && isDesktopUi) {
```

Comment:
I think we can show this even if it's not the desktop UI.

Reply:
Hmm, I was in understanding that when on mobile we would be triggering the native sharing UI, of which the design in still pending. Please correct me if I'm wrong!

Reply:
[Some mobile browsers](https://caniuse.com/#feat=web-share) trigger native sharing, but not all.  For those that don't, they'll get this UI!